### PR TITLE
Multiple plots

### PR DESCRIPTION
**Pull request by laispc**
Generalization of the structure of TimePlot in order to allow multiple curves to be plotted.
A sample of this feature was included in examples folder.

TimePlot continues to work the same way as before, but a method 'addYChannel' was created to add a curve to the plot. This method is used to add the curve for the PV defined in 'channel' property and also can be called in a Display class to add other curves.

Properties for curves besides the first one are not expose in QtDesigner, because it was not possible to create them dinamically. E.g. it is okay to turn yChannels into an array using type QStringList, but QtDesigner do not support an array of curveColor (there is not an array of QColor).

### CODE REVIEW

File: pydm/widgets/timeplot.py [hunk 1]
```diff
@@ -81,24 +80,31 @@ def configure_timer(self):
   
   # Adds a new curve to the current plot
   def addYChannel(self, ychannel='', curve_color=None):
-    # Allocate space for a new data buffer
-    self.num_of_channels = self.num_of_channels + 1
-    new_channel_id = self.num_of_channels
+    if ychannel == '':
+      raise Exception('[Error] Parameter ychannel should be defined for a new curve.')
```
Comment (hhslepicka) [inline pydm/widgets/timeplot.py]: What about changing the generic Exception to ValueError?
https://docs.python.org/3/library/exceptions.html#ValueError

File: pydm/widgets/timeplot.py [hunk 2]
```diff
@@ -81,24 +80,31 @@ def configure_timer(self):
...
+    # Add curve
+    if ychannel == self._ychannel:
+      new_curve = self.curve
+    else:
+      if curve_color is None:
+        raise Exception('[Error] Parameter curve_color should be defined for a new curve.')
```
Comment (hhslepicka) [inline pydm/widgets/timeplot.py]: Same as above

**Comment by mattgibbs** [comment]
Laís, this is looking like a really nice improvement.  If you are willing, I think it might be possible to expose this in Qt Designer, by having a second string list with a string representation of the color.  PyQtGraph's 'Color' class supports using single character strings representing color (similar to MATLAB, http://www.pyqtgraph.org/documentation/style.html#line-fill-and-color).  Additionally, you could let users pass in HTML style hex colors, like '#FF0000'.

(I also agree with Hugo's comments above, ValueError is probably more appropriate.)

**Comment by mattgibbs** [comment]
Hi Laís, I'm going to take your branch and add the color list functionality above.  Hopefully you haven't started working on it...

**Comment by laispc** [comment]
Matt and Hugo, thank you so much for the suggestions, I agree with them all. :)
A QStringList of colors that allows single char and hex values would be very nice. Channels could be put in a list also. Having both ychannels and colors in lists, addYChannel method may be simplified.
Matt, no problem, feel free to work on colors. I haven't done this yet.

**Comment by mattgibbs** [comment]
Hi All, I'm closing this pull request because I have opened #60, which builds on the foundation you guys made in this one.

**Pull request closed without merging**